stack.hpp

```diff
@@ -7,6 +7,13 @@ class Stack {
 public:
     Stack(std::size_t capacity): storage_(new T[capacity]), capacity_(capacity) {}
 
+    ~Stack(){
+        if(storage_ != nullptr) {
```

Comment:
Nice, C++11!

FYI: The if is not really required, delete[] nullptr does nothing and is save to call.